addons/mail/static/src/discuss/core/web/messaging_menu_patch.xml

```diff
@@ -3,14 +3,14 @@
     <t t-inherit="mail.MessagingMenu.content" t-inherit-mode="extension">
         <xpath expr="//div[hasclass('o-mail-MessagingMenu-header')]" position="inside">
             <div class="flex-grow-1"/>
-            <t t-if="ui.isSmall and (store.discuss.activeTab !== 'main' or !env.inDiscussApp)">
-                <div t-if="state.addingChat or state.addingChannel" class="w-100 p-1 bg-light" t-ref="selector">
+            <t name="searchBar" t-if="ui.isSmall and (store.discuss.activeTab !== 'main' or !env.inDiscussApp)">
+                <div t-if="displayChannelSelector" class="w-100 p-1 bg-light" t-ref="selector">
                     <t t-if="state.addingChat" t-set="category" t-value="store.discuss.chats"/>
                     <t t-if="state.addingChannel" t-set="category" t-value="store.discuss.channels"/>
                     <ChannelSelector category="category" autofocus="true"/>
                 </div>
             </t>
-            <button t-if="!ui.isSmall or (ui.isSmall and store.discuss.activeTab !== 'channel' and !state.addingChat and !(env.inDiscussApp and store.discuss.activeTab === 'main'))" t-att-class="ui.isSmall ? 'w-100 p-2 btn btn-secondary m-1' : 'btn btn-link p-2'" t-on-click.stop="onClickNewMessage">
+            <button t-if="!ui.isSmall or (ui.isSmall and store.discuss.activeTab !== 'channel' and store.discuss.activeTab !== 'whatsapp' and !state.addingChat and !(env.inDiscussApp and store.discuss.activeTab === 'main'))" t-att-class="ui.isSmall ? 'w-100 p-2 btn btn-secondary m-1' : 'btn btn-link p-2'" t-on-click.stop="onClickNewMessage">
```

Comment:
`whatsapp` is a technical string that shouldn't appear in mail module.

Either the condition should be changed, or an override should be made.

What are the tabs/channel type that we want to target here and what is different from the previous version?